### PR TITLE
Added dotenv to gemfile

### PR DESCRIPTION
**Pull request by bbrock25**
### _Why_?
I'm pretty sure that we agreed on using [dotenv](https://github.com/bkeepers/dotenv) for loading environment variables into a project. 

### _How_?
* Added dotenv-rails to Gemfile
* Alphabetized all gems in gemfile excluding rails, postgres and puma
* Added an example.env for formatting reference

### _Side Effects_?
* environment variables defined in .env will be available to your
  project

### CODE REVIEW

**Comment by Mrjaco12** [comment]
:+1:

File: Gemfile [hunk 1]
```diff
@@ -6,43 +6,42 @@ ruby '2.2.3'
...
-gem 'slim-rails'
-gem 'interactor'
+gem 'analytics-ruby', '~> 2.0.0', require: 'segment/analytics'
 gem 'awesome_print'
-
-gem 'bourbon'
 gem 'bitters'
+gem 'bourbon'
+gem 'coffee-rails'
+gem 'interactor'
```
Comment (samcdavid) [inline Gemfile]: we should use `interactor-rails` instead.

Comment (bbrock25) [inline Gemfile]: I didn't add that here, just tidy'd up a bit. the only gem i added was dotenv-rails. also, as an aside, I'm not sure that every project needs interactors.

Comment (samcdavid) [inline Gemfile]: It showed an addition so that is why I made the comment. Also, would it be more beneficial to already have it here and then if the team on the projects determines that it isn't needed they can remove it at a later time? I feel that having our preferred gems for those situations already included would be helpful so that we aren't using many different tools for the same functionality. That way we have a standardized tool set to make it easier to transition between projects should the need arise.

Comment (Mrjaco12) [inline Gemfile]: The logic for including it here is that the majority of our apps will be using controllers and if you're using a controller then please use interactors for organizing any controller logic. So the majority of our apps should probably be using them.

Comment (bbrock25) [inline Gemfile]: I'm not gonna change this PR for that, should be a separate PR

**Comment by jasontwong** [comment]
:+1:

File: .env.example [hunk 2]
```diff
@@ -0,0 +1,3 @@
+FOO=bar
+BIZ=baz
+
```
Comment (squaresurf) [inline .env.example]: I wonder if this should actually have meaning such as some dev vars like `RACK_ENV=development` or the like.

Comment (bbrock25) [inline .env.example]: I wonder if we have anything in the default project that requires an environment variable? maybe we could set up the sane segment configuration and update this file to have a default fake key.

**Comment by Mrjaco12** [comment]
Yeah, my vote would be update this file with good defaults as the relevant associated defaults code is included.

**Comment by samcdavid** [comment]
```
APPLICATION_ENVIRONMENT=development
AWS_ACCESS_KEY="found at https://spartanaws.signin.aws.amazon.com/"
AWS_FOG_DIRECTORY="found at https://spartanaws.signin.aws.amazon.com/"
AWS_SECRET_ACCESS_KEY="found at https://spartanaws.signin.aws.amazon.com/"
RACK_ENV=development
REDIS_URL=redis://localhost:6379/
ROLLBAR_ACCESS_TOKEN="found at https://rollbar.com/Spartan/CLIENT/"
ROLLBAR_CLIENT_ACCESS_TOKEN="found at https://rollbar.com/Spartan/CLIENT/"
SECRET_KEY_BASE=`bundle exec rake secret`
SEGMENT_KEY="found at https://segment.com/spartansystems/CLIENT/keys"
SMTP_ADDRESS=smtp.mandrillapp.com
SMTP_AUTHENTICATION=login
SMTP_DOMAIN=localhost
SMTP_ENABLE_STARTTLS_AUTO=true
SMTP_PASSWORD="found at https://mandrillapp.com/settings/"
SMTP_PORT=587
SMTP_USER_NAME=admin@client.com
```

**Comment by bbrock25** [comment]
I personally think that it suffices to have fabricated defaults right now. Once we have real dependencies in this app, we should replace these FOO and BAZ variables to reflect real ones. Lets just get this thing through :smile_cat: :rocket: help.
